src/script/ceph-backport.sh

```diff
@@ -132,7 +142,42 @@ then
 fi
 
 title=$(curl --silent 'http://tracker.ceph.com/issues/'$issue.json?key=$redmine_key | jq .issue.subject | tr -d '\\"')
-echo "Issue title: $title" 
+echo "Issue title: $title"
+
+function prepare () {
+    related_issue=$(curl --silent http://tracker.ceph.com/issues/$issue.json?include=relations |
+                    jq '.issue.relations[] | select(.relation_type | contains("copied_to")) | .issue_id')
+    [ -z "$related_issue" ] && echo "Could not find original issue." && return 1
+    echo "Original issue: $related_issue"
+
+    pr=$(curl --silent http://tracker.ceph.com/issues/$related_issue.json |
+         jq '.issue.custom_fields[] | select(.id | contains(21)) | .value' |
+         tr -d '\\"')
+    [ -z "$pr" ] && echo "Could not find PR." && return 1
+    echo "Original PR: $pr"
+
+    number=$(curl --silent 'https://api.github.com/repos/ceph/ceph/pulls/'$pr | jq .commits)
+    [ -z "$number" ] && echo "Could not determine the number of commits." && return 1
+    echo "Found $number commit(s)"
+
+    git fetch $ceph_repo
+    echo "fetch latest $milestone branch."
+
+    git checkout $ceph_repo/$milestone -b wip-$issue-$milestone
+
+    git fetch $ceph_repo pull/$pr/head:pr-$pr
+
+    git cherry-pick -x pr-$pr~$number..pr-$pr
+    echo "cherry picked pr-$pr into wip-$issue-$milestone"
+
+    return 0;
+}
+
+if [[ $* == *--prepare* ]]
+then
+    echo "--prepare found, will only prepare the backport"
+    prepare
```

Comment:
does execution end here, or does the call to `prepare` return and execution continues?

Reply:
It is meant to end, not sure how my manual test didn't catch that...
Now that the `prepare` function calls `exit 0`, the script will end after `prepare`.